plugins/importexport/native/filter/IssueNativeXmlFilter.inc.php

```diff
@@ -214,15 +210,13 @@ function addArticles($doc, $issueNode, $issue) {
 	 * @param $issue Issue
 	 */
 	function addIssueGalleys($doc, $issueNode, $issue) {
-		$filterDao = DAORegistry::getDAO('FilterDAO'); /* @var $filterDao FilterDAO */
-		$nativeExportFilters = $filterDao->getObjectsByGroup('issuegalley=>native-xml');
-		assert(count($nativeExportFilters)==1); // Assert only a single serialization filter
-		$exportFilter = array_shift($nativeExportFilters);
-		$exportFilter->setDeployment($this->getDeployment());
+		$currentFilter = PKPImportExportFilter::getFilter('issuegalley=>native-xml', $this->getDeployment());
 
 		$issueGalleyDao = DAORegistry::getDAO('IssueGalleyDAO'); /* @var $issueGalleyDao IssueGalleyDAO */
 		$issue = $issueGalleyDao->getByIssueId($issue->getId());
-		$issueGalleysDoc = $exportFilter->execute($issue);
+		$issueGalleysDoc = $currentFilter->execute($issue);
+
+		// throw new Exception('Ass Issue Galleys Exception');
 		if ($issueGalleysDoc->documentElement instanceof DOMElement) {
```

Comment:
What condition would cause this not to be true? Off the top of my head it would be an `issue_galley` node that didn't conform to the schema. In that case, rather than continuing without including it, it would be better for the export to fail.

Reply:
Because the initial code "https://github.com/pkp/pkp-lib/blob/master/classes/filter/Filter.inc.php#L442" returned the `$this->_output`, that check there indented not to try to add the `$issueGalleysDoc`. But you have a point there - Perhaps any such failure should result in a total process failure. 

If so, I will have to check in other parts too for such a behavior (I found it in 4-5 more places)

Reply:
Is this resolved?

Reply:
Resolved